optuna/__init__.py

```diff
@@ -20,3 +22,29 @@
 from optuna.study import Study  # NOQA
 from optuna.trial import Trial  # NOQA
 from optuna.version import __version__  # NOQA
+from optuna.type_checking import TYPE_CHECKING  # NOQA
+
+
+if TYPE_CHECKING:
+    from optuna import dashboard  # NOQA
+    from typing import Any  # NOQA
+else:
+
+    class _LazyImport(types.ModuleType):
+        def __init__(self, name):
+            # type: (str) -> None
```

Comment:
As discussed in #711, we've been using the function annotation syntax after dropping Python 2.

Reply:
Thank you for your review.
On the way to addressing the problem, I found that this class is not subject to type-checking.
Additionally, I had to import `typing` module even when `optuna.type_checking.TYPE_CHECKING` is False. The same will apply for `optuna.integration`.
Do you have any idea for a workaround, or are we okay with them?

Reply:
IMO, it's okay.
However, if anyone is concerned about the above points, we might be able to create, for example, `optuna._lazy_import` module and put the `_LazyImport` class under the module.